src/zc/queue/_queue.py

```diff
@@ -235,32 +235,23 @@ def __iter__(self):
                 yield i
 
     def __getitem__(self, index):
+        # not efficient, but quick and easy
         if isinstance(index, slice):
-            start, stop, stride = index.indices(len(self))
-            res = []
-            stride_ct = 1
-            for ix, v in enumerate(self):
-                if ix >= stop:
-                    break
-                if ix < start:
-                    continue
-                stride_ct -= 1
-                if stride_ct == 0:
-                    res.append(v)
-                    stride_ct = stride
-            return res
+            # We do this the easy way instead of attempting to handle
+            # negative strides ourself
+            return list(self)[index]
```

Comment:
I think you could preserve the laziness by using `return list(itertools.islice(self, *index.indices(len(self))))`.

And then I look at those four closing parens and think maybe that's not such a good idea after all.

Maybe `return list(itertools.islice(self, index.start, index.stop, index.step))`?  That does look better.